api_test.go

```diff
@@ -37,6 +27,12 @@ func WithServer(t *testing.T, f func(string)) {
 		time.Sleep(50 * time.Millisecond)
 	}
 
+	// Need to assign here since we need to call NewProxy to set the client.
+	testProxy.Name = "mysql_master"
+	testProxy.Listen = "localhost:3310"
+	testProxy.Upstream = "localhost:20001"
+	testProxy.Enabled = true
```

Comment:
Perhaps, it's not really ideal either. I'll take a look at what other HTTP wrappers do.